pkg/adminconfig/soft_policies.md

```diff
@@ -0,0 +1,105 @@
+Soft policies and infrastructure attributes
+===========================================
+
+## Summary
+
+Soft policies are used for data plane optimization. A soft policy focuses on one or several infrastructure metrics, e.g. storage cost or bandwidth between two regions.
+It provides a general directive such as minimize or maximize, and a relative weght given the workload and dataset information.  
+
+### Infrastructure attributes
+
+Infrastructure attributes are defined in `infrastructure.json` in /tmp/adminconfig directory. 
+
+```
+type InfrastructureElement struct {
+	// Attribute name, e.g. storage-cost, defined in the taxonomy
+	Attribute   taxonomy.Attribute    `json:"attribute"`
+	// Description
+	Description string                `json:"description,omitempty"`
+	// Value type (numeric, string, boolean)
+	Type        AttributeType         `json:"type"`
+	// Attribute value
+	Value       string                `json:"value"`
+	// Units (defined in the taxonomy)
+	Units       taxonomy.Units        `json:"units,omitempty"`
+	// A resource defined by the attribute ("storageaccount","module","cluster")
+	Object 		string				  `json:"object,omitempty"`
+	// A reference to the resource instance, e.g. storage account name
+	Instance    string                `json:"instance,omitempty"`
+	// A scale of values (minimum and maximum) when applicable
+	Scale       *adminrules.RangeType `json:"scale,omitempty"`
+	// A list of arguments defining a specific metric, e.g. regions for a bandwidth
+	Arguments   []string			  `json:"arguments,omitempty"`
+}
+```
+
+Attribute examples:
+
+```
+{
+    "attribute": "storage-cost",
+    "description": "theshire object store",
+    "value": "90",
+    "type": "numeric",
+    "units": "US Dollar",
+    "object": "storageaccount",
+    "instance": "account-theshire"
+}
+
+{
+    "attribute": "bandwidth",
+    "description": "bandwidth between neverland and theshire",
+    "value": "5",
+    "type": "numeric",
+    "units": "Megabit/sec",
```

Comment:
A tiny comment:
the ISO bandwidth abbreviations are:
Mbp/s, Gbp/s, Tbp/s for megabits, gigabits, terabits per second
and
MB/s, GB/s, TB/s for megabytes, gigabytes, and terabytes per sec.

Reply:
Thanks, changed.